lib/exfile/ecto/validate_content_type.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Exfile.Ecto.ValidateContentType do
+  alias Ecto.Changeset
+  alias Exfile.Processor.ContentType, as: ContentTypeProcessor
+
+  @registered_types %{
+    image: ~w(image/bmp image/gif image/jpeg image/png),
+    video: ~w(video/quicktime video/mp4 video/x-msvideo video/x-ms-wmv video/x-flv video/3gpp),
+    audio: ~w(audio/mpeg audio/x-wav audio/ogg)
+  }
+
+  def validate_content_type(changeset, field, accepted_content_types, message \\ "invalid format") do
+    case Changeset.get_change(changeset, field) do
+      %Exfile.File{} -> perform_validation(changeset, field, accepted_content_types, message)
+      _              -> changeset
+    end
+  end
+
+  defp perform_validation(changeset, field, accepted_content_types, message) do
+    { :ok, processed_file } = changeset
+      |> Changeset.get_change(field)
+      |> ContentTypeProcessor.call([], [])
+
+    changeset_with_processed_file = changeset
+      |> Changeset.put_change(field, processed_file)
```

Comment:
I found error here!

We are getting `%Exfile.File{}` struct here, but putting `%Exfile.Localfile{}` to changeset back! This is the reason why things like https://github.com/keichan34/exfile/blob/master/lib/exfile/phoenix/helpers.ex#L76 can't pattern match